patch.go

```diff
@@ -505,6 +508,22 @@ func (d *partialArray) remove(key string) error {
 
 }
 
+func (d *partialArray) removeByValue(key *lazyNode) error {
+	idx := -1
+	for i, v := range *d {
+		if key.equal(v) {
+			idx = i
+			break
+		}
+	}
+
+	if idx == -1 {
+		return errors.Wrapf(ErrInvalidIndex, "value not found")
+	}
+
+	return d.remove(strconv.Itoa(idx))
+}
```

Comment:
I'd normally write:
```
func (d *partialArray) removeByValue(key *lazyNode) error {
	for i, v := range *d {
		if key.equal(v) {
			return d.remove(strconv.Itoa(i))
		}
	}
	return errors.Wrapf(ErrInvalidIndex, "value not found")
}
```
Is this `idx`+`break` pattern some convention in the code here that you're following for consistency?